sympy/combinatorics/perm_groups.py

```diff
@@ -1643,6 +1644,25 @@ def contains(self, g, strict=True):
             return True
         return bool(self.coset_factor(g.array_form, True))
 
+    @property
+    def is_perfect(self):
+        """Return ``True`` if the group is perfect.
+        A group is perfect if it equals to its derived subgroup.
+
+        Examples
+        ========
+
+        >>> from sympy.combinatorics import Permutation
+        >>> from sympy.combinatorics.perm_groups import PermutationGroup
+        >>> a = Permutation(1,2,3)(4,5)
+        >>> b = Permutation(1,2,3,4,5)
+        >>> G = PermutationGroup([a, b])
+        >>> G.is_perfect
+        False
+
+        """
+        return self == self.derived_subgroup()
```

Comment:
This could be used to set `_is_perfect`. Then there would be no need to recompute the derived subgroup.

Reply:
I didn't get you doing this `_is_perfect = self.derived_subgroup()` will make it better?

Reply:
If you set `_is_perfect` when 'is_perfect` is called for the first time, it is not necessary to recompute the derived subgroup.

    if self._is_perfect is None:
        set self._is_perfect
    return self._is_perfect

Reply:
So, no need to compute derived_subgroup at all.

Reply:
@jksuom it shows error:
```
    set (self._is_perfect)
TypeError: 'NoneType' object is not iterable
```

Reply:
That was not valid Python code but pseudocode for the actual assignment of the proper value to `self._is_perfect`. For instance, if the proper value is `self == self.derived_subgroup()`, then that should be assigned.
`self._is_perfect` will the function as a cache. Computing `self.derived_subgroup()` is time consuming and that should not be repeated several times for asking if the group is perfect.

Reply:
Okay, that's why I got confused that how come we don't need to use `derived_subgroup()` at all and directly make is_perfect True.